staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go

```diff
@@ -42,3 +42,16 @@ func SetFeatureGateDuringTest(tb testing.TB, gate featuregate.FeatureGate, f fea
 		}
 	}
 }
+
+// InitFeatureGateDuringTest adds the specified Feature to the given FeatureGate. Failure to add the feature gate will cause
+// the test to fail. Use this function when the default feature gates are not automatically loaded.
+func InitFeatureGateDuringTest(tb testing.TB, gate featuregate.FeatureGate, f featuregate.Feature) {
```

Comment:
hmm... rather than test-only feature registration methods, I expected either a local feature gate object with just the relevant features set up, or plumbing feature enablement to the places it is used so it can be driven by unit tests without mutating global state.

Reply:
cc @deads2k

Reply:
> hmm... rather than test-only feature registration methods, I expected either a local feature gate object with just the relevant features set up, or plumbing feature enablement to the places it is used so it can be driven by unit tests without mutating global state.

Yeah, I would suggest having the controller itself have local `bool` values that are initialized during construction by passing the feature gates through.  This allows really clear code and makes unit tests easy to write.  We've done this in a few admission plugins.

Reply:
That makes sense, I'll update. I think this will come up in future refactors though and it would be convenient to have it. We're already mutating global state with `SetFeatureGateDuringTest` so it doesn't seem like a far stretch to initialize it as well.

Reply:
SetFeatureGateDuringTest was a bandaid to try to fix bad unit tests that were mutating global state and not restoring it. I'm not in favor of expanding the number of ways we do that.

Reply:
Good point. Had to set the feature gate fields as part of `Run()` since the unit tests rely on the constructor and running `Enabled` on the feature gate panics since the gate is never initialized for tests. I think ideally it should in the constructor but that would require a much larger refactor of the controller.

Reply:
Nevermind, did the full refactor now.